sdks/java/core/src/main/java/org/apache/beam/sdk/values/RowWithGetters.java

```diff
@@ -122,6 +123,15 @@ private Iterable getIterableValue(FieldType elementType, Object fieldValue) {
                   cacheKey, i -> getMapValue(type.getMapKeyType(), type.getMapValueType(), map))
           : (T) getMapValue(type.getMapKeyType(), type.getMapValueType(), map);
     } else {
+      if (type.isLogicalType(OneOfType.IDENTIFIER)) {
+        OneOfType oneOfType = type.getLogicalType(OneOfType.class);
+        OneOfType.Value oneOfValue = (OneOfType.Value) fieldValue;
+        Object convertedOneOfField =
+            getValue(oneOfValue.getFieldType(), oneOfValue.getValue(), null);
+        return (T)
```

Comment:
This is because the current semantics of Row.getValue() is to return the underlying type and you're expected to call Row.getLogicalTypeValue() to get the LogicalType value. It's debatable whether that's the best solution for Row, but there are arguments both ways and I didn't want to change those semantics here.